internal/util/util.go

```diff
@@ -398,8 +398,11 @@ func GetVolumeSnapshotRestoreWithStatusData(restoreName string, PVCName string,
 		}
 
 		if len(vsrList.Items) > 0 {
-			if vsrList.Items[0].Status.Phase == "Failed" {
-				return false, errors.Errorf("volumesnapshotrestore %v has failed status", vsrList.Items[0].Name)
+			if vsrList.Items[0].Status.Phase != "" {
+
+				if vsrList.Items[0].Status.Phase == "Failed" {
```

Comment:
Isn't
```
			if vsrList.Items[0].Status.Phase != "" {
				if vsrList.Items[0].Status.Phase == "Failed" {
```
the same as 
```
				if vsrList.Items[0].Status.Phase == "Failed" {
```
because `x == "Failed" != ""`

Did you make other changes? I don't think this is the PR that made the e2e pass.

Reply:
@kaovilai  Was e2e ran a few times since? It's possible the pass was a fluke, in which case I'll investigate more